komodo/release_transpiler.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+from argparse import ArgumentError, ArgumentParser, ArgumentTypeError
+from komodo import load_yaml, write_to_file, prettified_yaml
+
+rhel7_builtins = {
+    "python" : {
+        "rhel6": {
+            "py27" : "2.7.17",
+            "py36" : "3.6.9"
+        },
+        "rhel7": {
+            "py27" : "2.7.14+builtin",
+            "py36" : "3.6.9+builtin"
+        }
+    },
+    "qt5" : {
+        "rhel6": {
+            "py27" : None,
+            "py36" : "5.11.13",
+        },
+        "rhel7": {
+            "py27" : None,
+            "py36" : "5.11.13+builtin",
+        }
+    }
+}
+
+
+def build_matrix_file(release_base, release_folder, builtins):
+    py27 = load_yaml("{}/{}-py27.yml".format(release_folder, release_base))
+
+    py36 = load_yaml("{}/{}-py36.yml".format(release_folder, release_base))
+
+    all_packages = set(py36.keys()).union(py27.keys())
+
+    compiled = {}
+
+    for p in all_packages:
+        if p in builtins:
+            compiled[p] = builtins[p]
+        elif py27.get(p) == py36.get(p):
+            compiled[p] = py27.get(p)
+        else:
+            compiled[p] = {"py27": py27.get(p), "py36": py36.get(p)}
+
+    write_to_file(compiled, "{}.yml".format(release_base), False)
+
+def _build(packages, py_ver, rhel_ver):
+    release_dict = {}
+    for p, versions in packages.items():
+        if rhel_ver in versions:
+            version = versions[rhel_ver][py_ver]
+        elif py_ver in versions:
+            version = versions[py_ver]
+        else:
+            version = versions
+
+        if version:
+            release_dict[p] = version
+    return release_dict
+
+
+def transpile_releases(release_base, release_folder, output_folder):
+    release_matrix = load_yaml("{}.yml".format(os.path.join(release_folder, release_base)))
+    for rhel_ver in ("rhel6", "rhel7"):
+        for py_ver in ("py27", "py36"):
+            release_dict = _build(release_matrix, py_ver, rhel_ver)
+            filename = "{rel}-{pyver}-{rhel_version}.yml".format(
+                rel=release_base, pyver=py_ver, rhel_version=rhel_ver
+            )
+            write_to_file(release_dict, os.path.join(output_folder, filename))
+
+
+def combine(args):
+    build_matrix_file(args.release_base, args.release_folder, rhel7_builtins)
+
+def transpile(args):
+    transpile_releases(args.release_base, args.release_folder, args.output_folder)
+
+
+
+def main():
+    parser = ArgumentParser(description="Build release files.")
```

Comment:
The arguments appear to be optional. But they are not.
```
$ komodo-transpiler combine
Traceback (most recent call last):
  File "/home/jonas/Work/Equinor/musketeer/bin/komodo-transpiler", line 11, in <module>
    load_entry_point('Komodo==1.0', 'console_scripts', 'komodo-transpiler')()
  File "/home/jonas/Work/Equinor/musketeer/lib/python3.8/site-packages/komodo/release_transpiler.py", line 127, in main
    args.func(args)
  File "/home/jonas/Work/Equinor/musketeer/lib/python3.8/site-packages/komodo/release_transpiler.py", line 79, in combine
    build_matrix_file(args.release_base, args.release_folder, rhel7_builtins)
  File "/home/jonas/Work/Equinor/musketeer/lib/python3.8/site-packages/komodo/release_transpiler.py", line 34, in build_matrix_file
    py27 = load_yaml("{}/{}-py27.yml".format(release_folder, release_base))
  File "/home/jonas/Work/Equinor/musketeer/lib/python3.8/site-packages/komodo/prettier.py", line 130, in load_yaml
    raise argparse.ArgumentTypeError("{} is not a valid file".format(filename))
argparse.ArgumentTypeError: None/None-py27.yml is not a valid file
```

Make them positional since they cannot possible have any default.

Reply:
made required